### ISSUE TITLE
Suggestion: Can we have a "rustlings" equivalent for Burn/CubeCL? (burnlings / cubeclings)

### ISSUE THREAD

**Issue by Chuhan-Mateo** [tracel-ai/burn#5171]
The learning curve for Burn and CubeCL can be a bit steep for newcomers(like me) due to the lack of learning resources for beginners, and reading docs would be a bit hard and boring and truly understanding the crate.

I think it would be amazing if the community could build an interactive, command-line-driven tutorial suite similar to rustlings (e.g., burnlings, cubeclings, or cubeklings).

The tool could simply borrow the existing rustlings CLI engine(change all the exercises and solutions) but focus entirely on guiding users to fix intentional Burn tensor or CubeCL kernel compilation errors(even including cubeK).

I'm just throwing this idea out here to see if anyone in the community with more engineering time/skills would be interested in benchmarking or bootstrapping this. I believe it would drastically lower the barrier to entry for the Tracel AI ecosystem!

**Comment by Bellman281** [comment]
@Chuhan-Mateo 
we are preparing a book on deep learning with rust ...
you may find the book examples here:
https://github.com/jhosein58/learning-burn

**Comment by Chuhan-Mateo** [comment]
@Bellman281  Thank you! This book looks amazing! But what I was looking for is a more practical, interactive way for learning the whole Tracel-ai ecosystem, as what the [rustlings](https://github.com/rust-lang/rustlings/) did. But many of the examples showed up in the [book](https://github.com/jhosein58/learning-burn) could be use as the exercise for the burnlings(or whatever it will be name), so is it possible that we could make this book to transform into a interactive rustlings-like exercise instead of just the book for reading? And also is there other book similars to learning-burn but for cubeCL and cubeK?

**Comment by antimora** [comment]
Nice idea, but it fits better as a community project or a GitHub Discussion than as an issue here. Closing.
